### DISCUSSION TITLE
Getting issue device-simple using source Code

### DISCUSSION THREAD

**Discussion by Neeraj-Nec** [edgexfoundry/community#264] [Q&A]
Hello Team, 
I am creating new device-simple device. by using source code. My all the other EdgeX services is running into container as follow 

![image](https://github.com/edgexfoundry/community/assets/50608116/0bc8b8d9-a875-4d59-bdac-72bdd6d68891)

I am getting the following issue after running the command ./device-simple -cp=consul.http://10.10.26.104:8500

![image](https://github.com/edgexfoundry/community/assets/50608116/d46d4566-4290-4a07-b1a8-b406e5906ae3)

Thanks 
Neeraj

**Comment by Neeraj-Nec** [comment]
Hello Team,
Resolved The above issue now getting the following issue 

![image](https://github.com/edgexfoundry/community/assets/50608116/e795de35-7476-49a7-b32c-9bdc134aaed1)
 
I think here my device service is trying to connect with the registry by command  http://edgex-core-consul:8500" but as i am running the device service by source Code so it should be 10.10.26.206 . So how to configure it in the code?

Thanks
Neeraj

**Reply by jiekechoo** [reply]
Just use the command `./device-simple -cp -d -o -r` in your develop host machine to connect to the registry service.

Here is the help options:
```
./device-simple -h
Usage: ./device-simple [options]
Server Options:
    -cp, --configProvider        Indicates to use Configuration Provider service at specified URL.
                                 URL Format: {type}.{protocol}://{host}:{port} ex: consul.http://localhost:8500
    -cc, --commonConfig          Takes the location where the common configuration is loaded from when
                                 not using the Configuration Provider
    -o, --overwrite              Overwrite configuration in provider with local configuration
                                 *** Use with cation *** Use will clobber existing settings in provider,
                                 problematic if those settings were edited by hand intentionally
    -cf, --configFile <name>     Indicates name of the local configuration file. Defaults to configuration.yaml
    -p, --profile <name>         Indicate configuration profile other than default
    -cd, --configDir             Specify local configuration directory
    -r, --registry               Indicates service should use Registry.
    -rsh, 
     --remoteServiceHosts 
          <host names>           Indicates that the service is running remote from the core EdgeX services and
                                 to use the listed host names to connect remotely. <host names> contains 3 comma seperated host names seperated by ','.
                                 1st is the local system host name, 2nd is the remote system host name and 3rd is the WebServer bind host name
                                 example: -rsh=192.0.1.20,192.0.1.5,localhost
    -d, --dev                    Indicates service to run in developer mode which causes Host configuration values to be overridden.
                                 with `localhost`. This is so that it will run with other services running in Docker (aka hybrid mode)
    -i, --instance                  Provides a service name suffix which allows unique instance to be created
                                    If the option is provided, service name will be replaced with "<name>_<instance>"

Common Options:
    -h, --help                   Show this message
```

**Comment by Neeraj-Nec** [comment]
Thanks @jiekechoo for Your response

After Running the command ./cmd -cp consul.http://10.10.26.206:8500 -d -o -r I am getting the following issue 

![image](https://github.com/edgexfoundry/community/assets/50608116/13a59007-a7fb-4123-a77b-b973d5bf2d4b)

Can U please suggest how to solve this issue.

Thanks 
Neeraj

**Reply by jiekechoo** [reply]
You should not use ip address in the command line. ONLY: `./device-simple -cp -d -o -r` , `-d` means `localhost` for dev.

**Comment by Neeraj-Nec** [comment]
Hi @jiekechoo ,

Resolved the above issue now getting the following issue 

![image](https://github.com/edgexfoundry/community/assets/50608116/349ed3f6-7b2c-4088-9dbd-6d9d3877e1e5)


Thanks 
Neeraj

**Reply by jiekechoo** [reply]
If you want to test the EdgeX Foundry, just use [docker-compose ](https://github.com/edgexfoundry/edgex-compose) in `no-secty` mode. Please DO NOT use the source code for your testing purpose.
